doc/specs/#492 - Default Terminal/spec.md

```diff
@@ -0,0 +1,248 @@
+1. If any portion of the delegation fails, we will progressively degrade back to a `conhost` style Win32+GDI UX and nothing will be different from before.
+
+The settings experience includes:
+- Configuration of the delegation operations:
+    - Locations:
+        - With the registry
+            - This is what's going to be available first and will remain available. We will progress to some or all of the below after.
+            - We will need to potentially add specifications to this to both the default profile (for new installations of Windows) or to upgrade/migration profiles (for users coming from previous editions of Windows) to enable the delegation process, especially if we put a copy of Windows Terminal directly into the box.
+        - Inside Windows Terminal
+            - Inside the new Settings UI, we will likely need a page that configures the delegation keys in `HKCU\Console` or a link out to the Windows Settings panel, should we manage to get the settings configurable there.
+        - Inside the console property sheet
+            - Same as for Terminal but with `comctl` controlsover XAML +/- a link to the Windows Settings panel
+        - Inside the Settings panel for Windows (probably on the developer settings page)
+            - The ultimate location for this is likely a panel directly inside Windows. This is the hardest one to accomplish because of the timelines of the Windows product. We may not get this in an initial revision, but it should likely be our ultimate goal.
+    - Operation:
+        - Specify paths/server IDs - This is the initial revision
+        - Offer a list of registered servers or discovered manifests from the app catalog - This is the ideal scenario where we search the installed app catalog +/- the COM catalog and offer a list of apps that conform to the contract in a drop-down.
+    
+- Configuration of the legacy console state:
+    - Since we could end up in an experience where the default launch experience gets you directly into Windows Terminal, we believe that the Terminal will likely need an additional setting or settings in the new Settings UI that will allow the toggling of some of the `HKCU\Console` values to do things like set/remove the legacy console state.
+
+Concerns:
+- State separation policy for Windows. I believe `HKCU\Console` is already specified as a part of the "user state" that should be mutable and carried forward on OS Swap, especially as we have been improving the OS swap experience.
+- Ability for MSIs/elevated scripts to stomp the Delegation keys
+    - This was a long time problem for default app registrations and was limited in our OS. Are we about to run down the same path?
+    - What is the alternative here? To use a protocol handler? To store this configuration state data with other protected state in a registry area that is mutable, but only ACL'd to the `SYSTEM` user like some other things in the Settings control panel?

```

Comment:
Even if we did have the settings app handle the registration, how do we have it set up both the console and Terminal? The path to the terminal application might be easy enough, but the delegated console is probably not something that we want any user knowing about, and let the Terminal control how it configures that value (right?)

Reply:
The way the discussion on Settings UI has been going, we're going to let people choose their package, effectively. If the package has both a console and terminal registration, we'll set both in there. If only a terminal one, we'll set it to the terminal one and it'll use the inbox console.

However, the implementation for it will have two keys in the registry to allow mix and match and we will make the UI more capable and explain it further later if we need to. Also we'll work on making it happen outside a package later if we must based on demand.